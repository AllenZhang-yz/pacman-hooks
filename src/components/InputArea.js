import React from "react";

const InputArea = ({ onInputSubmit, handleChange, inputCommand }) => {
  return (
    <form className="input-area" onSubmit={e => onInputSubmit(e)}>
      <input
        placeholder="Please enter command"
        onChange={e => handleChange(e.target.value)}
        value={inputCommand}
      />
    </form>
  );
};

export default InputArea;
