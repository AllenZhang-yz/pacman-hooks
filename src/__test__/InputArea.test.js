import React from "react";
import InputArea from "../components/InputArea";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
Enzyme.configure({ adapter: new Adapter() });

const setup = ({ ...props }) => {
  const wrapper = shallow(<InputArea {...props} />);
  return {
    props,
    wrapper
  };
};

describe("InputArea", () => {
  const mock_handleChange = jest.fn();
  const mock_onInputSubmit = jest.fn();

  const { props, wrapper } = setup({
    handleChange: (mock_handleChange),
    onInputSubmit: (mock_onInputSubmit),
    inputCommand: "test command"
  });
  it("InputArea renders correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("InputArea When inputvalue is passed", () => {
    const input = wrapper.find('input');
    expect(input.props().value).toBe(`${props.inputCommand}`);
  })
  it("InputArea onChange called correctly", () => {
    const input = wrapper.find('input');
    const value = props.inputCommand;
    input.simulate('change', { target: { value }});
    expect(mock_handleChange).toBeCalled();
    expect(mock_handleChange).toBeCalledWith(value);
  })
  it("InputArea onSubmit called correctly", () => {
    const form = wrapper.find('form');
    form.simulate('submit');
    expect(mock_onInputSubmit).toBeCalled();
  })
});
