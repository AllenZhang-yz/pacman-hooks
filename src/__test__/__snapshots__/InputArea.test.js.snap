// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputArea InputArea renders correctly 1`] = `
<form
  className="input-area"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Please enter command"
    value="test command"
  />
</form>
`;
